### PR TITLE
sig-release: Rearrange dash configs and prune some jobs from sig-release-job-config-errors

### PR DESCRIPTION
**Pull request by justaugustus**
This continues the pruning work I started in https://github.com/kubernetes/test-infra/pull/15039.

Here we're doing a few things:
- Add sig-release-generated dashboard to allow us to easily identify generated tests
- ~Allow kubernetes-bazel-build jobs to fork into release dashes by moving them to *-informing~
- Drop kops and Windows jobs from job-config-errors dash
- Drop GCE/GKE jobs from job-config-errors dash

The remaining jobs on the `sig-release-job-config-errors` are in [`config/jobs/kubernetes/sig-cli/sig-cli-config.yaml`](https://github.com/kubernetes/test-infra/blob/3c66bac45a151d8eb3a6908035da0b0d5fc1a374/config/jobs/kubernetes/sig-cli/sig-cli-config.yaml).

Several of them seem to be passing, but are incorrect named e.g., `1.12-1.13` when they're using something like stable1/stable2 markers
There are enough of them configured in confusing ways, that I'd like to tackle them in a separate PR.

### CODE REVIEW

File: config/jobs/kubernetes/sig-testing/bazel-build-test.yaml [hunk 1]
```diff
@@ -123,35 +123,6 @@ postsubmits:
       testgrid-dashboards: sig-release-master-informing
       testgrid-tab-name: bazel-build-master
       description: 'OWNER: sig-testing; Builds kubernetes at each bommit using bazel with RBE'
-  - name: ci-kubernetes-bazel-test # TODO(fejta): delete after 1m (dup of post-kubernetes-bazel-test)
```
Comment (justaugustus) [inline config/jobs/kubernetes/sig-testing/bazel-build-test.yaml]: Dropping this as @fejta left the `TODO` in May.

**Comment by justaugustus** [comment]
cc: @alejandrox1 @alenkacz @kubernetes/sig-release-admins @kubernetes/release-team

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.14.yaml [hunk 2]
```diff
@@ -369,7 +369,7 @@ periodics:
 postsubmits:
   kubernetes/kubernetes:
   - annotations:
-      testgrid-dashboards: sig-release-job-config-errors
+      testgrid-dashboards: google-unit, sig-release-1.14-informing
```
Comment (cblecker) [inline config/jobs/kubernetes/sig-release/release-branch-jobs/1.14.yaml]: I think this job is misnamed on L379

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.15.yaml [hunk 3]
```diff
@@ -383,7 +383,7 @@ periodics:
 postsubmits:
   kubernetes/kubernetes:
   - annotations:
-      testgrid-dashboards: sig-release-job-config-errors
+      testgrid-dashboards: google-unit, sig-release-1.15-informing
     branches:
```
Comment (cblecker) [inline config/jobs/kubernetes/sig-release/release-branch-jobs/1.15.yaml]: I'm confused.. are these postsubmits or periodics?

**Comment by justaugustus** [comment]
(dropped the bazel job changes to make this less controversial.)
/assign @Katharine @cblecker

**Comment by justaugustus** [comment]
Thanks Katharine!
/hold cancel

**Pull request merged**